main.tf

```diff
@@ -657,8 +657,140 @@ resource "helm_release" "castai_kvisor_self_managed" {
+            }
+          }
+        }
+      }
+
+      dynamic "cluster_limits" {
+        for_each = try(autoscaler_policy_overrides.value.cluster_limits, null) != null ? [autoscaler_policy_overrides.value.cluster_limits] : []
+
+        content {
+          enabled = try(cluster_limits.value.enabled, null)
+
+
+          dynamic "cpu" {
+            for_each = try(cluster_limits.value.cpu, null) != null ? [cluster_limits.value.cpu] : []
+
+            content {
+              min_cores = try(cpu.value.min_cores, null)
+              max_cores = try(cpu.value.max_cores, null)
+            }
+          }
+        }
+      }
+
+      dynamic "spot_instances" {
+        for_each = try(autoscaler_policy_overrides.value.spot_instances, null) != null ? [autoscaler_policy_overrides.value.spot_instances] : []
+
+        content {
+          enabled                             = try(spot_instances.value.enabled, null)
+          max_reclaim_rate                    = try(spot_instances.value.max_reclaim_rate, null)
+          spot_diversity_enabled              = try(spot_instances.value.spot_diversity_enabled, null)
+          spot_diversity_price_increase_limit = try(spot_instances.value.spot_diversity_price_increase_limit, null)
+
+          dynamic "spot_backups" {
+            for_each = try(spot_instances.value.spot_backups, null) != null ? [spot_instances.value.spot_backups] : []
+
+            content {
+              enabled                          = try(spot_backups.value.enabled, null)
+              spot_backup_restore_rate_seconds = try(spot_backups.value.spot_backup_restore_rate_seconds, null)
+            }
+          }
+
+          dynamic "spot_interruption_predictions" {
+            for_each = try(spot_instances.value.spot_interruption_predictions, null) != null ? [spot_instances.value.spot_interruption_predictions] : []
+
+            content {
+              enabled                            = try(spot_interruption_predictions.value.enabled, null)
+              spot_interruption_predictions_type = try(spot_interruption_predictions.value.spot_interruption_predictions_type, null)
+            }
+          }
+        }
+      }
+
+      dynamic "node_downscaler" {
+        for_each = try(autoscaler_policy_overrides.value.node_downscaler, null) != null ? [autoscaler_policy_overrides.value.node_downscaler] : []
+
+        content {
+          enabled = try(node_downscaler.value.enabled, null)
+
+          dynamic "empty_nodes" {
+            for_each = try(node_downscaler.value.empty_nodes, null) != null ? [node_downscaler.value.empty_nodes] : []
+
+            content {
+              enabled       = try(empty_nodes.value.enabled, null)
+              delay_seconds = try(empty_nodes.value.delay_seconds, null)
+            }
+          }
+
+          dynamic "evictor" {
+            for_each = try(node_downscaler.value.evictor, null) != null ? [node_downscaler.value.evictor] : []
```

Comment:
why we cannot have this ` for_each = try(node_downscaler.value.evictor, [])`?

Reply:
because of the `[node_downscaler.value.evictor]` transformation in then statement

Reply:
but double check if TF is handing this magically. If it can, going to replace it

Reply:
` for_each = try([autoscaler_policy_overrides.value.unschedulable_pods], [])` this works

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNDdjdWY1MnI0YWJhZGlreTA0M3dxajRlMTQybm15Ymd5aGxpOTgyMyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/SXHmIKdu0Jci25nHz7/giphy.gif)